src/main/Main.hs

```diff
@@ -574,6 +579,18 @@ uploadCmd args go = do
                 (tarName, tarBytes) <- getSDistTarball pkgDir
                 liftIO $ Upload.uploadBytes uploader tarName tarBytes
 
+sdistCmd :: [String] -> GlobalOpts -> IO ()
+sdistCmd [] _ = error "To create an sdist tarball for the current package, please run 'stack sdist .'"
```

Comment:
In the `sdist` case, I think defaulting to current directory is fine. This *is* in contradiction to `stack upload`, since the latter has major side-effects.